### PR TITLE
Return empty byte array

### LINKED ISSUE #203: Empty blob should return byte[0]
A column with type BLOB and length = 0 should return `byte[0]` and `IsDBNull == false`.

Instead, SqliteDataReader returns `null` and `IsDBNull == false`.

Repro: https://gist.github.com/natemcmaster/6f2ef57338c597ccfa65

### PR DESCRIPTION
**Pull request by natemcmaster**
fix #203

### CODE REVIEW

File: src/Microsoft.Data.Sqlite/Interop/NativeMethods.cs [hunk 1]
```diff
@@ -91,7 +91,7 @@ public static byte[] sqlite3_column_blob(Sqlite3StmtHandle pStmt, int iCol)
             var ptr = Sqlite3.column_blob(pStmt, iCol);
             if (ptr == IntPtr.Zero)
             {
-                return null;
+                return new byte[0];
```
Comment (davidfowl) [inline src/Microsoft.Data.Sqlite/Interop/NativeMethods.cs]: `Array.Empty<byte>()`

Comment (natemcmaster) [inline src/Microsoft.Data.Sqlite/Interop/NativeMethods.cs]: `.Empty<T>()` doesn't exist .net < 4.6 (Mono)

Comment (davidfowl) [inline src/Microsoft.Data.Sqlite/Interop/NativeMethods.cs]: ifdef?

**Comment by bricelam** [comment]
Does similar behavior occur for empty strings? Are NULL and an empty array the same in C? Why is the native API returning NULL?

File: src/Microsoft.Data.Sqlite/Interop/NativeMethods.cs (hunk 1 above)
Comment (bricelam) [inline src/Microsoft.Data.Sqlite/Interop/NativeMethods.cs]: Boo. I hate #if-in just to use Array.Empty. If caching is really an issue, cache it statically on this type.

**Comment by natemcmaster** [comment]
> Strings returned by sqlite3_column_text() and sqlite3_column_text16(), even empty strings, are always zero-terminated. The return value from sqlite3_column_blob() for a zero-length BLOB is a NULL pointer.

https://www.sqlite.org/c3ref/column_blob.html.

[Zero-length arrays](https://gcc.gnu.org/onlinedocs/gcc/Zero-Length.html) in C depend on the version of C.

**Comment by bricelam** [comment]
:shipit: Thanks for following up.

**Comment by bricelam** [comment]
Does `SqliteDataReader.IsDBNull` return `true` For these values?

**Comment by natemcmaster** [comment]
This is the original issue. This fixes this misalignment of SqliteDataReader returning `null` even though `IsDBNull == false`.

**Pull request merged**